doc/src/contributing/cross-referencing.rst

```diff
@@ -0,0 +1,110 @@
+.. _style_guide_cross-referencing:
+
+Cross-Referencing
```

Comment:
This doesn't really make sense as a top-level page. It makes more sense to put it as a section in the docstrings page.